src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.CachedDocumentSearch.cs

```diff
@@ -104,40 +109,35 @@ private static void ClearCachedData()
             var (patternName, patternContainer) = PatternMatcher.GetNameAndContainer(searchPattern);
             var declaredSymbolInfoKindsSet = new DeclaredSymbolInfoKindSet(kinds);
 
-            await SearchCachedDocumentsInCurrentProcessAsync(
+            await foreach (var item in SearchCachedDocumentsInCurrentProcessAsync(
                 storageService, patternName, patternContainer, declaredSymbolInfoKindsSet,
-                onItemFound, priorityDocumentKeys, cancellationToken).ConfigureAwait(false);
+                priorityDocumentKeys, cancellationToken).WithCancellation(cancellationToken))
```

Comment:
@jcouv do i need to call .WithCancellation in all these plaes?  i'm annotating all `async IAsyncEnumerable` methods with the appropriate `[EnumeratorCancellation]` attribute, so i'm hping the answer is 'no'.

Reply:
@CyrusNajmabadi That's correct. `.WithCancellation(...)` isn't needed when the method is annotated with `[EnumeratorCancellation]` and you're passing the cancellation token.
http://blog.monstuff.com/archives/2019/03/async-enumerables-with-cancellation.html